mycore-base/src/main/java/org/mycore/datamodel/niofs/MCRBasicFileAttributesProxy.java

```
@@ -0,0 +1,98 @@
+package org.mycore.datamodel.niofs;
+
+import java.nio.file.attribute.FileTime;
+
+import org.mycore.common.digest.MCRDigest;
+
+/**
+ * Provides a lazy loading implementation for {@link MCRFileAttributes}.
+ *
+ * <p>The attributes are loaded only once on the first request and are then cached for subsequent uses.</p>
+ *
+ * <p>This approach is particularly beneficial in remote filesystems where network latency and data transfer costs
+ * are significant. In some scenarios, instances of this proxy might be created without ever accessing the underlying
+ * attributes, thus avoiding unnecessary network traffic entirely.</p>
+ *
+ * @param <T> The type of the file key.
+ * @param <P> The type of the {@link MCRPath} associated with these file attributes.
+ */
+public abstract class MCRBasicFileAttributesProxy<T, P extends MCRPath> implements MCRFileAttributes<T> {
+
+    private volatile MCRFileAttributes<T> proxy;
+
+    private final Object lock = new Object();
+
+    protected final P path;
+
+    public MCRBasicFileAttributesProxy(P path) {
+        this.path = path;
+    }
```

This is not how attributes work. Call to the methods should not result in exceptions once they are constructed:
`Files.readAttributes(file, MyFileAttributes.class)` should throw any exception and not a call to lastAccessTime(). Often a check for existence of a file is implemented as reading the file attributes.

---

Ok, will rm this class entirely and solve the attribute loading in ocfl